pyupgrade/_plugins/versioned_branches.py

```diff
@@ -188,5 +188,10 @@ def visit_If(
     ):
         if node.orelse and not isinstance(node.orelse[0], ast.If):
             yield ast_to_offset(node), _fix_py3_block_else
-        elif not node.orelse:
+        elif not node.orelse and min_version == (3, 0):
+            # allow >= (3,) and > (3,) to be dedented even without
+            # an else clause, as a special cases
             yield ast_to_offset(node), _fix_py3_block
+        else:
+            # don't try to fix, e.g., >= (3, 6) if there's no else clause
+            yield ast_to_offset(node), lambda i, tokens: None
```

Comment:
we shouldn't put noop callbacks into the callback machinery -- I think the original logic was fine here (?)

Reply:
actually the result with the original logic for that one broken test is
```diff
import sys
if True:
-    if sys.version_info >= (3, 6):
-        pass
+   pass
```
so that sounds about right to me, I guess this test was misplaced and shouldn't have been in the "noop" zone at all ?
I'll push to revert this.